src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1785,6 +1785,13 @@ where
 
     for path in paths {
         let path = path.as_ref();
+
+        // check if path points to $CARGO_HOME and skip to avoid rebuilds when only caching
+        // $CARGO_HOME/registry/{index, cache} and $CARGO_HOME/git/db
+        // this lines up with discussions in #9455 about making $CARGO_HOME readonly
```

Comment:
I feel like there is some assumed context in this comment, like what caching is happening.

Maybe something like:

> Assuming anything in `cargo_home` is immutable (see also #9455 about marking it readonly) which avoids rebuilds when CI caches `$CARGO_HOME/registry/{index, cache}` and `$CARGO_HOME/git/db` across runs, keeping the content the same but changing the mtime.

Reply:
that does look better indeed. Updated